src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs

```diff
@@ -74,7 +74,39 @@ public void GetErrorCode_SingleSonarIssue_ErrorCodeReturned(string fullRuleKey,
         }
 
         [TestMethod]
-        public void GetErrorCode_NonStandardErrorCode_NoException_ErrorCodeNotReturned()
+        [DataRow("S666", "csharpsquid", "S666", "SonarAnalyzer.CSharp")]
+        [DataRow("S666", "vbnet", "S666", "SonarAnalyzer.VisualBasic")]
+        [DataRow("S234", "vbnet", "S234", "SonarAnalyzer.VisualBasic")]
+        [DataRow("c:S111", "c", "S111", "SonarLint")]
+        [DataRow("cpp:S222", "cpp", "S222", "SonarLint")]
+        [DataRow("javascript:S333", "javascript", "S333", "SonarLint")]
+        [DataRow("typescript:S444", "typescript", "S444", "SonarLint")]
+        [DataRow("secrets:S555", "secrets", "S555", "SonarLint")]
+        [DataRow("foo:bar", "foo", "bar", "SonarLint")]
+        public void TryGetRuleId_FromHandle_ErrorCodeReturned(string fullRuleKey, string expectedRepo, string expectedRule, string buildTool)
+        {
+            // Note: this is a copy of TryGetRuleIdFromSelectedRow_SingleSonarIssue_ErrorCodeReturned,
+            //       but without the serviceProvider and IErrorList setup
```

Comment:
Exactly the same as the existing test, but with less setup.
I couldn't think of a good alternative to just copying the existing tests - let me know if you can.

Reply:
Hmm, I cant really think of  a better way either